python/tests/nonreg/arcticdb/version_store/test_nonreg_specific.py

```diff
@@ -294,4 +296,117 @@ def test_date_range_multi_index(lmdb_version_store):
     result_df = lib.read(
         sym, date_range=DateRange(pd.Timestamp("2099-01-01"), pd.Timestamp("2099-01-02"))
     ).data
-    assert_frame_equal(result_df, expected_df)
\ No newline at end of file
+    assert_frame_equal(result_df, expected_df)
+
+
+@pytest.mark.parametrize(
+    "method",
+    ("write", "append", "update", "write_metadata", "batch_write", "batch_append", "batch_write_metadata")
+)
+@pytest.mark.parametrize("lib_config", (True, False))
+@pytest.mark.parametrize("env_var", (True, False))
+@pytest.mark.parametrize("arg", (True, False, None))
+def test_prune_previous_general(version_store_factory, method, lib_config, env_var, arg):
+    lib = version_store_factory(prune_previous_version=lib_config, use_tombstones=True)
+    should_be_pruned = lib_config
+    if env_var:
+        os.environ["PRUNE_PREVIOUS_VERSION"] = "true"
```

Comment:
`monkeypatch.setenv` to make sure we don't interfere with other tests

Reply:
Done